Team9Project/src/com/indragie/cmput301as1/OnlineManager.java

```diff
@@ -0,0 +1,104 @@
+/* 
+ * Copyright (C) 2015 Jimmy Ho
+ * 
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *  
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.indragie.cmput301as1;
+
+import java.io.IOException;
+import java.util.LinkedList;
+
+import com.indragie.cmput301as1.ElasticSearchAPIClient.APICall;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.util.Log;
+
+public class OnlineManager<T extends ElasticSearchDocument> extends BroadcastReceiver{
+
+	private LinkedList<ElasticSearchAPIClient.APICall<T>> stack = new LinkedList<ElasticSearchAPIClient.APICall<T>>(); 
+	
+	public void enqueueCall(ElasticSearchAPIClient.APICall<T> call) {
+		stack.add(call);
+		checkForNetwork();
+	}
+	
+	private Context context;
+
+	public void checkForNetwork() {
+		// if there is something inside of the stack, do this, else do nothing
+		if (stack.size() > 0) {	
+			
+			if (isNetworkAvailable(context) == true) {
```

Comment:
Don't need explicit comparison to true, `if (isNetworkAvailable(context)) {` to stay consistent with code style.